webapp/app/common/basic-schemas.ts

```diff
@@ -9,3 +9,11 @@ export const hexSchema = hexChars.transform((str) => str as Hex);
 export const addressSchema = hexChars
   .refine((str) => str.length === 42, "Address have to be 20 bytes long")
   .transform((str) => getAddress(str));
+
+export const numericStrSchema = z.coerce
+  .bigint({ message: "Must be a integer number" })
+  .transform((num) => num.toString());
+
+export const nonZeroNumericStrSchema = numericStrSchema.refine((s) => !/^0+$/.test(s), {
+  message: "Cannot be zero.",
+});
```

Comment:
Can we do this check in the same way as the other? Transforming to number then back?

Reply:
Done